lib/java/src/org/apache/thrift/transport/TFramedTransport.java

```diff
@@ -123,6 +123,12 @@ public void consumeBuffer(int len) {
     readBuffer_.consumeBuffer(len);
   }
 
+  public void clear() {
+    if (readBuffer_ != null) {
```

Comment:
(I'm the person who originally wrote this patch / feature request)
I agree, and can't find any reflected use cases in at least just the thrift codebase.